backend/src/dao/ShoutoutsDao.ts

```diff
@@ -76,6 +76,20 @@ export default class ShoutoutsDao extends BaseDao<Shoutout, DBShoutout> {
     return this.createDocument(shoutoutWithUUID.uuid, shoutoutWithUUID);
   }
 
+  /**
+   * Edits a shoutout
+   * @param uuid - uuid of the shoutout
+   * @param shoutout - shoutout object
+   */
+  async editShoutout(uuid: string, newMessage: string): Promise<Shoutout> {
+    await this.collection.doc(uuid).update({ message: newMessage });
+    const updatedDoc = await this.getDocument(uuid);
+    if (!updatedDoc) {
+      throw new Error('Failed to fetch updated shoutout...');
```

Comment:
Remove the `...` in the string?

Reply:
Also, could you use `updateShoutout` instead of creating a method just to edit the message?

Reply:
This is more for future reference, but I think checking if the document exists in the DAO layer might be repetitive, since we already check if the uuid exists in the API layer. Not including the checks here also helps keep the DAO files consistent, since most of them do not have additional checks.

Reply:
woah thanks for catching this @patriciaahuang, not sure how I didn't see it. We should not do any permissions checks at the DAO layer. That's reserved for the API layer!